refactor: move to atlas

Vilsol opened the pull request.

mircearoata commented on db/schema/mixin_time.go:
Do we want these to be nullable, or is it just to match the existing schema?

Vilsol replied:
Not entirely sure why I set them to be nullable, prod DB has no fields with them being NULL

mircearoata reviewed (changes requested):
Both the GQL API and the REST mod all versions endpoint now return the mod ID in the mod_id field, which breaks SMM (both v2 and v3) and ficsit-cli (SMM v2 because it specifically uses getModByReference, and SMM v3/ficsit-cli because the lockfile will contain mod IDs instead of references for dependencies, but mod references for profile items).

For the custom index names on some of the types, do we actually need those? Since atlas didn't try to recreate them, it means they would simply get renamed, like the deleted_at indexes, so we'll have less custom indexes for stuff that's just a single-field unique.

Vilsol commented on db/schema/version.go:
```diff
@@ -24,11 +25,11 @@ func (Version) Fields() []ent.Field {
 		field.String("mod_id"),
 		field.String("version").MaxLen(16),
-		field.String("game_version"),
+		field.String("game_version").Optional(), // TODO Should be backfilled and turned non-optional
```
We can, but then we have 2 problems:

1. Many mods are with invalid constraints: https://gist.github.com/Vilsol/7e4fbcc59b0f4327f64382aac9d8d86a
2. What should we backfill the version with? `0`?

Vilsol replied:
I have changed it so it gets backfilled with `0`'s

Vilsol commented on db/schema/user.go:
```diff
@@ -27,18 +27,18 @@ func (User) Fields() []ent.Field {
-		index.Fields("google_id"),
-		index.Fields("facebook_id"),
+		index.Fields("email").Unique().StorageKey("uix_users_email"),
+		index.Fields("github_id").StorageKey("idx_users_github_id"),
+		index.Fields("google_id").StorageKey("idx_users_google_id"),
+		index.Fields("facebook_id").StorageKey("idx_users_facebook_id"),
```
They were never unique in the first place, that was a mistake. Unique index would require them being set for every user.

mircearoata replied:
That wouldn't be the case, duplicate null values are considered distinct (at least by postgres), unless `NULLS NOT DISTINCT` is used on the index. https://www.postgresql.org/docs/current/indexes-unique.html#INDEXES-UNIQUE

Vilsol commented:
I reverted back the change so `mod_id` now will return the `mod_reference` as before, added a deprecation notice to the field, and added a new `mod_reference` field that should be transitioned to as soon as this hits prod.
